docs/ordinals/ft.md

```diff
@@ -0,0 +1,181 @@
+---
+sidebar_position: 3
+---
+
+# Funglible Tokens - FTs
+
+Just like NFTs, `scrypt-ts` also supports fungible tokens. Under the hood it utilizes the [`bsv-20` protocol](https://docs.1satordinals.com/bsv20).
+
+Currently, the SDK supports both v1 and v2 of the `bsv-20` FT protocol.
+
+
+## v1
+
+Tokens utilizing the first version of the `bsv-20` must be initialized by a **deployment** inscription, which specifies the tokens ticker symbol, amount and mint limit. For more information refer to the [1Sat docs](https://docs.1satordinals.com/bsv20#v1-mint-first-is-first-mode).
+
+
+To create a v1 token smart contract, have it extend the `BSV20V1` class:
+
+```ts
+class HashPuzzleFT extends BSV20V1 {
+    @prop()
+    hash: Sha256
+
+    constructor(tick: ByteString, max: bigint, lim: bigint, hash: Sha256) {
+        super(tick, max, lim)
+        this.init(...arguments)
+        this.hash = hash
+    }
+
+    @method()
+    public unlock(message: ByteString) {
+        assert(this.hash == sha256(message), 'hashes are not equal')
+    }
+}
+```
+
+As you can see above, the constructor of contract extending the `BSV20V1` takes as parameters all of the needed information for the token deployment, succeeded by other parameters needed you use for your contract (`hash` in this particular example).
+
+
+### Deployment
+
+Here's an example of how you would deploy the new FT:
+
+```ts
+HashPuzzleFT.loadArtifact();
+
+const tick = toByteString("DOGE", true);
+const max = 100000n;
+const lim = max / 10n;
+
+const hashPuzzle = new HashPuzzleFT(
+  tick,
+  max,
+  lim,
+  sha256(toByteString("secret0", true))
+);
+await hashPuzzle.connect(getDefaultSigner());
+await hashPuzzle.deployToken();
+```
+
+### Mint and Transfer
+
+Once the deployment transaction was successfully broadcast, the token is ready for minting.
+
+Here's how you can mint some amount:
+
+```ts
+// Minting
+const amt = 1000n;
+const mintTx = await hashPuzzle.mint(amt);
+console.log("Minted tx: ", mintTx.id);
+```
+
+Note, that if the amount exceeds the limit set above, or the token was already wholely minted, the transaction won't be considered valid by 1Sat indexers.
+
+The minted amount can then be transferred by unlocking the mint transaction:
```

Comment:
by calling the contract, as in regular sCrypt contracts:

Link to https://docs.scrypt.io/how-to-deploy-and-call-a-contract/#contract-call